### PR TITLE
Add string_manipulation.md challenge.

### PR DESCRIPTION
**Pull request by scox03**
This PR meets the requirements of the string_manipulation.md challenge.

### CODE REVIEW

File: shania_cox/alpha_challenges/string_manipulation.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+```ruby
+"IIcpIcpmIIhIhImsatIaatImiInatAaYkwIsImBiietttaIetsiuwyjtapWsihaIstbtfpawihtapImlwtapnagnagnagwtapNagNnNtgtIgotadfapeglimlNtgbwtapmImhsitAwtapIkIstbtIkIstlbeIktAinInsiayklidmaIkiiIdIhdBwtapmWawtaWtapm"
+```
+
+Programmatically manipulate `iverson_rant` to reflect the output above.
+
+```ruby
+words = iverson_rant.split(" ")
+words.each do |word|
+  print word[0]
```
Comment (gretchenziegler) [inline shania_cox/alpha_challenges/string_manipulation.md]: This is one way to approach it, but I don't think you're getting the exact output you want. When I run your code in pry, `print`ing does print the acronym, but the original `words` array is what is returned (plus a weird little modified bonus):
```
IIcpIcpmIIhIhImsatIaatImiinataaYkwIsImbiietttaIetsiuwyjtapWsihaIstbtfpawihtapImlwtapnagnagnagwtapNagNnNtgtIgotadfapegli
=> ["If",pmImhsitAwtapIkIstbtIkIstlbeIktaInInsiayklidmaIkiiIdIhdBwtapmWawtaWtapm
 "I",
 "can't",
 "practice,",
 "I",
 "can't",
 "practice,",
 "man.",
 "If",
 "I'm",
 "hurt,", etc.
```
How would you return _just_ that acronym? Here's a hint: look into `.map` and `.join`.

Comment (adi89) [reply shania_cox/alpha_challenges/string_manipulation.md]: ruby has a nify `#chr` method, which will return the first character of your string.

**Comment by scox03** [comment]
This pull request has revisions for the string_manipulation.md challenge

File: shania_cox/alpha_challenges/string_manipulation.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+```ruby
+count = 0
+words = iverson_rant.split
+while i < words.length
+  if words[i] == 'practice,' || words[i] == 'practice.'
+    count += 1
+  end
+  i += 1
+end
+print count
```
Comment (gretchenziegler) [inline shania_cox/alpha_challenges/string_manipulation.md]: What happens when you run this code using pry or irb? It's always a good idea to test your code before you push it, to make sure there aren't any unforeseen errors 😄 

Also, now that you've modified the first exercise to be more compact, is there anything similar that you could do here?

File: shania_cox/alpha_challenges/string_manipulation.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+* Replace the spaces with underscores.
+* Replace the newline (`\n`) characters with `"*"`.
+* Replace all instances of 'Sam' with your name.
+* Change all instance of the letter 'o' to the number 0.
+* Change the word 'green' to 'blue'.
+
+Hint: You can chain substitutions!
+
+```ruby
+sam_i_am.gsub(" ", "_").gsub("\n", "*").gsub("Sam", "Shania").gsub("o", "0").gsub("green", "blue")
```
Comment (gretchenziegler) [inline shania_cox/alpha_challenges/string_manipulation.md:48]: 👍

**Comment by scox03** [comment]
This PR shows an attempt to condense my response to question 2 via the map method and by incorporating a loop. I did test this with the irb and it gave me an appropriate answer but the array was also printed too with nil values wherever practice was not located. I'm not entirely sure if that should be there though.

File: shania_cox/alpha_challenges/string_manipulation.md [hunk 4]
```diff
@@ -26,16 +26,7 @@ print words
...
-iverson_rant.split.map {|word| 
-  case word
-  when "practice,"
-    count += 1
-  when "practice."
-    count +=1
-  end
-}
-print count
+puts iverson_rant.scan("practice").count
```
Comment (gretchenziegler) [inline shania_cox/alpha_challenges/string_manipulation.md:29]: Nice! ⚡

**Review by gretchenziegler — APPROVED** [review]

**Pull request merged**